### PR TITLE
cancel context for each scenario

### PR DESCRIPTION
**Pull request by draganm**
### 🤔 What's changed?

Each Scenario gets a cancellable context that is cancelled at the end of the scenario.

### ⚡️ What's your motivation? 

Cancelled context can be used to shut down mock services that are run in the scope of one Scenario (see #513)

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by draganm** [comment]
Not sure how to end-to-end test this though.

**Review by mirogta — COMMENTED** [review]
This looks great. Would you mind adding a simple example code as well as part of this PR? I would prefer if each use case (like your new one, however it looks cool) has an appropriate example to follow otherwise some less experienced developers might be a bit lost in how to use it.

**Comment by vearutop** [comment]
I think this is a good idea, although it potentially can break existing scenarios that depend (implicitly or explicitly) on background tasks after the scenario has ended already. 
In such (rare?) cases, context cancellation can be canceled 🙃 with help of a simple wrapper.

File: suite.go [hunk 1]
```diff
@@ -447,6 +447,12 @@ func (s *suite) runPickle(pickle *messages.Pickle) (err error) {
 		ctx = context.Background()
 	}
 
+	var cancel context.CancelFunc
+
+	ctx, cancel = context.WithCancel(ctx)
```
Comment (vearutop) [inline suite.go:452]:
```suggestion
	ctx, cancel := context.WithCancel(ctx)
```

**Comment by vearutop** [comment]
Actually, I think this is good to merge, and I'll add a test in a follow-up commit.

**Pull request merged**